packages/grafana-ui/src/components/Table/utils.ts

```diff
@@ -385,10 +385,19 @@ export function getFooterItems(
 }
 
 function getFormattedValue(field: Field, reducer: string[], theme: GrafanaTheme2) {
-  const fmt = field.display ?? getDisplayProcessor({ field, theme });
   const calc = reducer[0];
-  const v = reduceField({ field, reducers: reducer })[calc];
-  return formattedValueToString(fmt(v));
+  const reducerInfo = fieldReducers.get(calc);
+  const format = field.display ?? getDisplayProcessor({ field, theme });
+  const fieldCalcValue = reduceField({ field, reducers: reducer })[calc];
+
+  // If the reducer preserves units then format the
+  // end value with the field display processor
+  if (reducerInfo.preservesUnits) {
```

Comment:
Just wanted to leave a comment here that this would be the area to refactor and pull out into a separate function in this part so we can use in other places (i.e. XY chart, etc.)